### PR TITLE
User, Group, Permission model specs

### PR DESCRIPTION
**Pull request by jgthibodeau**

### CODE REVIEW

File: spec/factories/group.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FactoryGirl.define do
+  factory :group do |f|
+    f.name 'group'
+    f.description 'this is a group'
```
Comment (dfaulken) [inline spec/factories/group.rb]: I see from your group test that names should be unique - should `name` be a sequence then? 

Also, you can define factories without the block variable, e.g.

```ruby
FactoryGirl.define do
  factory :group do
    sequence(:name){ |n| "Group #{n}" }
    description 'Description'
  end
end
```

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -19,10 +19,80 @@
...
+    # params[:controller], params[:action], params[:id]
+    it 'returns true if the user has a permission with the requested controller, action, and no id_field' do
+      user = create(:user)
+      group = create(:group)
+      permission = create(:permission, controller: 'user', action: 'show', id_field: nil)
+
+      group.permissions << permission
+      user.groups << group
+
+      expect(user.has_permission?(permission.controller, permission.action, nil)).to eq true
```
Comment (dfaulken) [inline spec/models/user_spec.rb]: For this and other predicate methods that start with `has_`, you should be able to use RSpec's dynamic predicate method matchers, e.g.:

```ruby
 expect(user).to have_permission permission.controller,
                                 permission.action,
                                 nil
```

Comment (dfaulken) [inline spec/models/user_spec.rb]: (Shameless plug: if you're interested in incorporating a RuboCop style check to make sure you're using dynamic predicate method matchers, [there's a gem for that](https://github.com/umts/custom-cops)!)

Comment (jgthibodeau) [inline spec/models/user_spec.rb]: That's pretty snazzy

**Pull request merged**